julia/GradBench/src/GradBench.jl

```diff
@@ -2,48 +2,73 @@ module GradBench
 
 import JSON
 
-const DISPATCH_TABLE = Dict{String,Dict{String,Any}}()
+"""
+    Experiment
 
-function register!(mod, functions)
+Defines a measurment experiment.
+
+## Interface
+- `args = preprocess(::Experiment, message)`
+- `ret = (::Experiment)(args...)`
+- Optional: `postprocess(::Experiment, ret)`
+"""
+abstract type Experiment end
+
+function preprocess(::Experiment, message)
+    return message
+end
+
+function postprocess(::Experiment, ret)
+    return ret
+end
+
+const DISPATCH_TABLE = Dict{String,Dict{String,Experiment}}()
+
+function register!(mod, experiments)
     if haskey(DISPATCH_TABLE, mod)
         error("mod $mod is already registered")
     end
-    DISPATCH_TABLE[mod] = functions
+    DISPATCH_TABLE[mod] = experiments
     return
 end
 
 # Avoid measuring dispatch overhead
-function measure(func::F, arg) where {F}
+function measure(func::F, args...) where {F}
     start = time_ns()
-    ret = func(arg)
+    ret = func(args...)
     done = time_ns()
     return ret, done - start
 end
```

Comment:
This looks a bit clumsy now that we use it several times, can we just use `@timed`?

Reply:
Ok judging by https://github.com/gradbench/gradbench/pull/514#discussion_r2049253355 we want the function barrier, nevermind